### PR TITLE
Adding Paired Interval Filter to Filter Sam Reads

### PR DESCRIPTION
**Pull request by kbergin**
Adds an interval filter to filter sam reads that filters over intervals while keeping paired reads together.

### CODE REVIEW

File: src/test/java/picard/sam/FilterSamReadsTest.java [hunk 1]
```diff
@@ -69,9 +90,51 @@ public void testJavaScriptFilters(final String samFilename, final String javascr
...
+                .setCreateIndex(true).makeBAMWriter(builder.getHeader(), false, inputSam);
+
+        for (final SAMRecord record : builder) {
+            writer.addAlignment(record);
+        }
+        writer.close();
+
+        final File intervalFile = new File(intervalFilename);
+
+        final FilterSamReads program = new FilterSamReads();
```
Comment (vivster7) [inline src/test/java/picard/sam/FilterSamReadsTest.java]: could refactor tests to make more DRY.

File: src/test/java/picard/sam/FilterSamReadsTest.java [hunk 2]
```diff
@@ -56,22 +77,70 @@ public void testJavaScriptFilters(final String samFilename, final String javascr
...
+            program.INTERVAL_LIST = inputFile;
+        }
+        else {
+            program.JAVASCRIPT_FILE = inputFile;
+        }
+
+        return program;
+    }
+
+    private int getReadCount(FilterSamReads filterTest) throws Exception {
```
Comment (vivster7) [inline src/test/java/picard/sam/FilterSamReadsTest.java]: I think this is a great function to java 8-ify (stream and count the iterator), if you want the practice

**Comment by vivster7** [comment]
👍

**Comment by nh13** [comment]
@kbergin branch is still here.

**Pull request merged**